tasks/tasks.go

```diff
@@ -0,0 +1,93 @@
+package tasks
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+// Status is the status of the task.
+type Status int
+
+// TaskExecutor executes a series of task functions.
+type TaskExecutor interface {
+	QueueTask(task func() error) string
+	GetTaskStatus(id string) (Status, error)
```

Comment:
This and it also allows more possible ids (since each character can be a number _or_ a letter).